include/aspect/stokes_matrix_free.h

```diff
@@ -0,0 +1,781 @@
+      for (unsigned int cell=0; cell<n_cells; ++cell)
+        for (unsigned int i=0; i<this->get_matrix_free()->n_components_filled(cell); ++i)
+          {
+
+            if (for_mg)
+              {
+                typename DoFHandler<dim>::level_cell_iterator FEQ_cell = this->get_matrix_free()->get_cell_iterator(cell,i);
+                typename DoFHandler<dim>::level_cell_iterator DG_cell(&tria,
+                                                                      FEQ_cell->level(),
+                                                                      FEQ_cell->index(),
+                                                                      &dof_handler_for_projection);
+                DG_cell->get_active_or_mg_dof_indices(local_dof_indices);
+              }
+            else
+              {
+                typename DoFHandler<dim>::active_cell_iterator FEQ_cell = this->get_matrix_free()->get_cell_iterator(cell,i);
+                typename DoFHandler<dim>::active_cell_iterator DG_cell(&tria,
+                                                                       FEQ_cell->level(),
+                                                                       FEQ_cell->index(),
+                                                                       &dof_handler_for_projection);
+                DG_cell->get_active_or_mg_dof_indices(local_dof_indices);
+              }
+
+            //TODO: projection with higher degree
+            Assert(local_dof_indices.size() == 1, ExcNotImplemented());
+            for (unsigned int q=0; q<velocity.n_q_points; ++q)
+              viscosity_x_2(cell,q)[i] = 2.0*viscosity_values(local_dof_indices[0]);
+          }
+    }
+
+    template <int dim, int degree_v, typename number>
+    void
+    ABlockOperator<dim,degree_v,number>
+    ::local_apply (const dealii::MatrixFree<dim, number>                 &data,
+                   dealii::LinearAlgebra::distributed::Vector<number>       &dst,
+                   const dealii::LinearAlgebra::distributed::Vector<number> &src,
+                   const std::pair<unsigned int, unsigned int>           &cell_range) const
+    {
+      FEEvaluation<dim,degree_v,degree_v+1,dim,number> velocity (data,0);
+
+      for (unsigned int cell=cell_range.first; cell<cell_range.second; ++cell)
+        {
+          AssertDimension(viscosity_x_2.size(0), data.n_macro_cells());
+          AssertDimension(viscosity_x_2.size(1), velocity.n_q_points);
+
+          velocity.reinit (cell);
+          velocity.read_dof_values(src);
+          velocity.evaluate (false, true, false);
+          for (unsigned int q=0; q<velocity.n_q_points; ++q)
+            {
+              velocity.submit_symmetric_gradient
+              (viscosity_x_2(cell,q)*velocity.get_symmetric_gradient(q),q);
+            }
+          velocity.integrate (false, true);
+          velocity.distribute_local_to_global (dst);
+        }
+    }
+    template <int dim, int degree_v, typename number>
```

Comment:
More empty lines between functions, go once through all the functions in this file and check that they are limited by empty lines.